tests/integration/snowflake/test_incremental_build.py

```diff
@@ -0,0 +1,152 @@
+            container.remove(force=True)
+
+            # Do a native (of temporarily "native" arch) build
+            runner.invoke(cli, ('create',
+                                '-i',
+                                f'snowshu_replica_integration-test-incremental:{fake_local_arch}',
+                                '--replica-file',
+                                CONFIGURATION_PATH),
+                                catch_exceptions=False)
+
+            # Check if all images are present
+            new_replicas_list = client.images.list(
+                'snowshu_replica_integration-test')
+            assert len(new_replicas_list) == 1
+
+            # Check if data is fine
+            container = client.containers.run(f'snowshu_replica_integration-test:{fake_local_arch}',
+                                              ports={'9999/tcp': 9999},
+                                              name='integration-test',
+                                              network='snowshu',
+                                              detach=True)
+            time.sleep(DOCKER_SPIN_UP_TIMEOUT)
+
+            assert get_tables(
+                'integration-test') == expected_incremented_table_list
+            container.remove(force=True)
+            
+            # Remove new images before building another incremental
+            for image in client.images.list('snowshu_replica_integration-test'):
+                for tag in image.tags:
+                    client.images.remove(tag)
+
+            # Do a multiarch incremental build
+            runner.invoke(cli, ('create',
+                                '-i',
+                                f'snowshu_replica_integration-test-incremental:{fake_local_arch}',
+                                '--replica-file',
+                                CONFIGURATION_PATH,
+                                '-m'),
+                                catch_exceptions=False)
+
+            new_replicas_list = client.images.list(
+                'snowshu_replica_integration-test')
+            assert len(new_replicas_list) == 2
+
+            # Check if all new replicas have correct set of tables
+            for replica in new_replicas_list:
+                container = client.containers.run(replica,
+                                                  ports={'9999/tcp': 9999},
+                                                  name='integration-test',
+                                                  network='snowshu',
+                                                  detach=True)
+                time.sleep(DOCKER_SPIN_UP_TIMEOUT)
+
+                assert get_tables(
+                    'integration-test') == expected_incremented_table_list
+                container.remove(force=True)
+
+            for image in client.images.list('snowshu_replica_integration-test'):
+                for tag in image.tags:
+                    client.images.remove(tag)
```

Comment:
it looks to me that this test is overloaded with logic, why don't to create a set of small tests?
benefit: easier to maintain

Reply:
It's all dependent on the order of execution. I could split it for regular and multiarch variants, but not much after that.